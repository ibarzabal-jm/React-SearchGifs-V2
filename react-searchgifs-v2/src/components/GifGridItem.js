import React from 'react'

export const GifGridItem = ( {id, title, url} ) => {

    

    return (
        <div className="card animate__animated animate__rollIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

