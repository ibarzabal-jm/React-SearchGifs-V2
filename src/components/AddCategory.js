import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    
    const [inputValue, setInputValue] = useState(''); 
    
    const handleInputChange = ( e ) =>{
        setInputValue( e.target.value );

    }

    const handleSubmit = ( e ) => {
        
        e.preventDefault();

        if( inputValue.trim().length > 2){

            setCategories( categorias => [inputValue, ...categorias ] );
            setInputValue('');
        }else{
            
            console.log('Categor√≠as no agregada')
        }

    }

    return (
        <>
            <h4>Ingresa aqui abajo lo que quieras buscar</h4>
            <form onSubmit={handleSubmit}>
                <input
                    type = "text"
                    value = {inputValue}
                    onChange={ handleInputChange }
    
                />

            </form>
        </>
    )
}


AddCategory.propType = {
    setCategories: PropTypes.func.isRequired
}

