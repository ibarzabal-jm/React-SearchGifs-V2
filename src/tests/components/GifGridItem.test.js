import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'



describe('Pruebas en <GrifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow( <GifGridItem title = {title} url = {url}/> );
    
    
    test('debe de mostrar el componente correctamente ', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(title);
    })
    
    test('debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        
        // console.log( img.props().src );
        expect( img.prop('src') ).toBe(url);
        expect( img.prop('alt') ).toBe(title);
        
    })
    
    test('debe de tener animate__rollIn', () => {
        
        const div = wrapper.find('div');
        
        const className = div.prop('className');

        expect( className.includes('animate__rollIn') ).toBe(true);
    })
    
    
})
