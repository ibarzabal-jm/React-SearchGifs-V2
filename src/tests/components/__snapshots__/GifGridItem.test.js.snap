// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GrifGridItem /> debe de mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__rollIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
